### ISSUE TITLE
unable to focus trap in ios safari 

### ISSUE THREAD

**Issue by Elecweb** [reactjs/react-modal#713]
### Summary:
focus trap is works fine in desktop browser but it doens't work on ios safari.
In ios safari, when you focus, for example, input on modal, you can click up/down arrow to focus another input which is outside of modal.

### Steps to reproduce:
 1. go to this link [https://codesandbox.io/s/n5ny70m2qp](https://codesandbox.io/s/n5ny70m2qp) on ios safari
 1. click "open Modal" button
 2. focus input on modal
 3. click up/down arrow and you'll see you still can focus on input which's not on modal

### Expected behavior:

you shouldn't focus any input/element outside modal's area

### Link to example of issue:
[https://codesandbox.io/s/n5ny70m2qp](https://codesandbox.io/s/n5ny70m2qp)

### Additional notes:
I tried with iPhone X, iPhon 8 Plus  in browserstack and real devices

![safariiosfocustrap](https://user-images.githubusercontent.com/13183413/47964896-67f80080-e072-11e8-9469-2630c3c7786a.gif)

**Comment by Elecweb** [comment]
after investigation, it look likes clicking up/down arrow for tabbing doesn't fire event at all
so `handleKeyDown` doesn't get called (so it doesn't call  `scopeTab` helper)
